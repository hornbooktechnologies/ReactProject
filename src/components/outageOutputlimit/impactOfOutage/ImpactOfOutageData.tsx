import React from "react";
import { useTranslation } from "react-i18next";
import { ImpactOfOutageDataModel } from "../../../models/outageOutputLimit/ImpactOfOutageDataModel";
import { cssClassName } from "../../../utils/CssConstants";
import Tooltip from "../../common/Tooltip";
import { OutageOutputConstant } from "../../../utils/AppConstants";
import { IsappearTooltip, prefixSign, getDiffOfValues } from "../../../utils/utils";

const ImpactOfOutageData = ({
  value,
  positiveVal,
  negativeVal,
  title,
  valuePrevious,
  positivePreviousVal,
  negetivePreviousVal,
  prefix,
  suffix,
}: ImpactOfOutageDataModel) => {
  const { i18n } = useTranslation();
  let negVariation = OutageOutputConstant.NEGATIVE_VAL
  let posVariation = OutageOutputConstant.POSITIVE_VAL
  let diffValue = getDiffOfValues(value,valuePrevious)
  let diffPositiveVal = getDiffOfValues(positiveVal,positivePreviousVal)
  let diffNegativeVal = getDiffOfValues(negativeVal,negetivePreviousVal)
  
  return (
    <>
      <div className="uk-card">
        <label className="font-18">{title}</label>
        <div className="mb-24">
          <span
            className={`${IsappearTooltip(diffValue)} font-28 uk-text-bold mt-0 mb-20 ${
              parseInt(value) <= 0
                ? `${cssClassName.COLOR_GREY_BLUE_70}`
                : `${cssClassName.COLOR_GREEN_60}
                  `
            }`}
          >
            {prefixSign(value, posVariation)}
            {value}
            <span
              className={`${
                diffValue === true ? "color-grey-blue-10" : ""
              } font-10 ${cssClassName.COLOR_WHITE} uk-text-top `}
            >
              {suffix}
            </span>
            {diffValue === true ? (
              <Tooltip
                prefix={prefixSign(valuePrevious, posVariation)}
                previousDayValue={valuePrevious}
                unit={suffix}
              />
            ) : null}
          </span>
        </div>
        <div
          className="uk-child-width-1-1@s uk-child-width-1-2@m uk-grid ml--25"
          uk-grid="true"
        >
          <div className="media-mb-20">
            <p
              className={`uk-margin-remove font-14 ${cssClassName.COLOR_GREEN_60}`}
            >
              {i18n.t("OUTAGE_AND_OUTPUT_LIMIT.INCREMENT_IMPACT")}
            </p>
            <span className={`mt-5 mb-0 uk-text-bold font-20 ${cssClassName.COLOR_WHITE}
                  ${IsappearTooltip(diffPositiveVal)}`}
                  >
              {prefix}
              {prefixSign(positiveVal,posVariation)}
              {positiveVal}
              <span className="font-10 uk-text-normal uk-text-top">
                {suffix}
              </span>
              {diffPositiveVal === true ? (
                <Tooltip
                  prefix={prefix + prefixSign(positivePreviousVal,posVariation)}
                  previousDayValue={positivePreviousVal}
                  unit={suffix}
                />
                ) : null}
              </span>
          </div>

          <div>
            <p className={`uk-margin-remove font-14 ${cssClassName.COLOR_RED_60}`}>
              {i18n.t("OUTAGE_AND_OUTPUT_LIMIT.DECREMENT_IMPACT")}
            </p>
            <span className={`mt-5 mb-0 uk-text-bold font-20 
                  ${cssClassName.COLOR_WHITE}
                  ${IsappearTooltip(diffNegativeVal)}`}
            >
              {prefix}
              {prefixSign(negativeVal,negVariation)}
              {negativeVal}
              <span className="font-10 uk-text-normal uk-text-top">
                {suffix}
              </span>

            {diffNegativeVal === true ? (
              <Tooltip
                prefix={prefix + prefixSign(negetivePreviousVal,negVariation)}
                previousDayValue={negetivePreviousVal}
                unit={suffix}
              />
            ) : null}
            </span>
          </div>
        </div>
      </div>
    </>
  );
};

export default ImpactOfOutageData;
