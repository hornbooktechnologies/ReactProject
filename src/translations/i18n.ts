import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
