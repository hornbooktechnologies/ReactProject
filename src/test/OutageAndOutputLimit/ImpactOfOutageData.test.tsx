import React from "react";
import renderer from "react-test-renderer";
import ImpactOfOutageData from "../../components/outageOutputlimit/impactOfOutage/ImpactOfOutageData";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";
describe("Component: ImpactOfOutageData", () => {
  it("ImpactOfOutageData renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
        <ImpactOfOutageData
          value={10}
          impactVal={10}
          impactVal1={10}
          title={"ImpactOfOutageDataTitle"}
        />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
