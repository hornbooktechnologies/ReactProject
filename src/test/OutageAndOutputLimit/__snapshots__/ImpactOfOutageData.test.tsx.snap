// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ImpactOfOutageData ImpactOfOutageData renders correctly 1`] = `
<div
  className="uk-card"
>
  <label
    className="font-18"
  >
    ImpactOfOutageDataTitle
  </label>
  <div
    className="mb-24"
  >
    <span
      className=" font-28 uk-text-bold mt-0 mb-20 color-green-60
                  "
    >
      +
      10
      <span
        className=" font-10 color-white uk-text-top "
      />
    </span>
  </div>
  <div
    className="uk-child-width-1-1@s uk-child-width-1-2@m uk-grid ml--25"
    uk-grid="true"
  >
    <div
      className="media-mb-20"
    >
      <p
        className="uk-margin-remove font-14 color-green-60"
      >
        OUTAGE_AND_OUTPUT_LIMIT.INCREMENT_IMPACT
      </p>
      <span
        className="mt-5 mb-0 uk-text-bold font-20 color-white
                  "
      >
        +
        <span
          className="font-10 uk-text-normal uk-text-top"
        />
      </span>
    </div>
    <div>
      <p
        className="uk-margin-remove font-14 color-red-60"
      >
        OUTAGE_AND_OUTPUT_LIMIT.DECREMENT_IMPACT
      </p>
      <span
        className="mt-5 mb-0 uk-text-bold font-20 
                  color-white
                  "
      >
        <span
          className="font-10 uk-text-normal uk-text-top"
        />
      </span>
    </div>
  </div>
</div>
`;
